kubechain/internal/controller/taskruntoolcall/taskruntoolcall_controller_test.go

```diff
@@ -95,6 +93,132 @@ var _ = Describe("TaskRunToolCall Controller", func() {
 			utils.ExpectRecorder(recorder).ToEmitEventContaining("ExecutionSucceeded")
 		})
 	})
+
+	Context("Pending -> Failed", func() {
+		It("fails when arguments are invalid", func() {
+			teardown := setupTestAddTool(ctx)
+			defer teardown()
+
+			// Create TaskRunToolCall with Pending status but invalid arguments
+			taskRunToolCall := &TestTaskRunToolCall{
+				name:      "invalid-args-trtc",
+				toolName:  addTool.name,
+				arguments: `invalid json`, // Invalid JSON
+			}
+			defer taskRunToolCall.Teardown(ctx)
```

Comment:
i would say techincally this should come after the SetupWithStatus

Reply:
Fixed in #44: https://github.com/humanlayer/kubechain/pull/44/files#diff-11cafbf859aa294cd3ca18f53fd8d9bdbc363ff83077d6df5ebb12c5b2a8e37dR118